src/CfgMgr32/CfgMgr32+CM_NOTIFY_EVENT_DATA.cs

```diff
@@ -0,0 +1,133 @@
+            /// the registration that generated this notification event data.
+            /// </summary>
+            [FieldOffset(0)]
+            public CM_NOTIFY_FILTER_TYPE FilterType;
+
+            /// <summary>
+            /// Reserved. Must be 0.
+            /// </summary>
+            [FieldOffset(4)]
+            public uint Reserved;
+
+            // DeviceInterface
+
+            /// <summary>
+            /// The GUID of the device interface class for the device interface to which the notification event data pertains.
+            /// </summary>
+            [FieldOffset(8)]
+            public Guid ClassGuid;
+
+            /// <summary>
+            /// The first character of the symbolic link path of the device interface to which the notification event data pertains.
+            /// Use <see cref="GetSymbolicLink(CM_NOTIFY_EVENT_DATA*, int)"/> to retrieve this value.
+            /// </summary>
+            [FieldOffset(24)]
+            public char SymbolicLink;
+
+            // DeviceHandle
+
+            /// <summary>
+            /// The GUID for the custom event.
+            /// </summary>
+            [FieldOffset(8)]
+            public Guid EventGuid;
+
+            /// <summary>
+            /// The offset of an optional string buffer. Usage depends on the contract for the EventGuid.
+            /// </summary>
+            [FieldOffset(24)]
+            public int NameOffset;
+
+            /// <summary>
+            /// The number of bytes that can be read from the Data member.
+            /// </summary>
+            [FieldOffset(28)]
+            public int DataSize;
+
+            /// <summary>
+            /// The first bit of optional binary data. Usage depends on the contract for the EventGuid.
+            /// </summary>
+            [FieldOffset(32)]
+            public byte Data;
+
+            // DeviceInstance
+
+            /// <summary>
+            /// The first character of the device instance ID of the device to which the notification event data pertains.
+            /// Use <see cref="GetInstanceId(CM_NOTIFY_EVENT_DATA*, int)"/> to retrieve this value.
+            /// </summary>
+            [FieldOffset(8)]
+            public char InstanceId;
+
+            /// <summary>
+            /// Gets the symbolic link path of the device interface to which the notification event data pertains.
+            /// </summary>
+            /// <param name="eventData">
+            /// The event notification.
+            /// </param>
+            /// <param name="eventDataSize">
+            /// The event notification size.
+            /// </param>
+            /// <returns>
+            /// The symbolic link path of the device interface to which the notification event data pertains
+            /// </returns>
+            public static string GetSymbolicLink(CM_NOTIFY_EVENT_DATA* eventData, int eventDataSize)
```

Comment:
Why not define these as instance methods so they don't have to take a pointer to themselves?